aiida/plugins/entry_point.py

```diff
@@ -30,11 +30,18 @@
 ENTRY_POINT_STRING_SEPARATOR = ':'
 
 
-@functools.lru_cache(maxsize=1)
+@functools.cache
 def eps():
     return _eps()
 
 
+@functools.lru_cache(maxsize=100)
```

Comment:
Are we worried about the size of the cache? I think the number of different calls to `eps_select` should be reasonable, not exceeding on the order of 100. So wouldn't we be better of with `lru_cache(maxsize=None)` i.e. `cache` which will be faster. Not sure how much faster it will be compared to an LRU cache of max 100 items. Might be negligible

Reply:
I am actually worried. I wouldn't be surprised if the number of calls was bigger then 100, especially if plugins are installed, since in some functions we're essentially iterating over all existing entry points when looking for the entry point for a given class. I'll take a closer look and do some more benchmarking.

Reply:
Are we doing this iteration or is it importlib?

Are we trying to find an entry point for a class without knowing the name of the entry point?

Reply:
> since in some functions we're essentially iterating over all existing entry points when looking for the entry point for a given class. I'll take a closer look and do some more benchmarking.

These don't call this `eps_select` function though, do they? The cache here simply applies to the number of combinations of arguments with which it is called. Since it just has `group` and `name`, it should just be the list of all `(group, name)` tuples with which the function is called. This should, reasonably, not be much larger than the entry points that exist.